Website/Default.aspx.cs

```diff
@@ -658,6 +658,11 @@ protected override void OnInit(EventArgs e)
                 {
                     var originalurl = Context.Items["UrlRewrite:OriginalUrl"].ToString();
                     CanonicalLinkUrl = originalurl.Replace(PortalSettings.PortalAlias.HTTPAlias, primaryHttpAlias);

```

Comment:
Is this the only location where `CanonicalLinkUrl` is being set? Why only when `primaryHttpAlias != null`?